### PR TITLE
Adding Performance Checklist to site

### PR DESCRIPTION
**Pull request by sekyondaMeta**
Adding performance checklist to site. Quick first run at the checklist.
Plan to add this checklist to FAQs page in the future as well. For now, it can be accessed through the performance page.

Fixes #(issue)
Adds to performance information

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Built pages locally to ensure they work.

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

**Review by msaroufim — APPROVED** [review]

File: docs/performance_checklist.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+- Use model quantization (i.e., int8) for CPU inference.  Explore different quantization options: dynamic quantization, static quantization, and quantization aware training, as well as tools such as Intel Neural Compressor that provide more sophisticated quantization methods.
+
+- Balance throughput and latency with smart batching.  While meeting the latency SLA try larger batch sizes to increase the throughput.
+
+- Try [torchscript](https://pytorch.org/docs/stable/jit.html), [inference_mode](https://pytorch.org/docs/stable/generated/torch.inference_mode.html), and [optimize_for_inference](https://pytorch.org/docs/stable/generated/torch.jit.optimize_for_inference.html). Torchscript provides tools that incrementally transition models from purely python to a torchscript program which can be run independently of python. This gives us the ability to run models in environments where python may be at a disadvantage in terms of performance.
```
Comment (msaroufim) [inline docs/performance_checklist.md:30]: this is quite outdated best to remove this line

File: docs/performance_checklist.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+As this [example](https://colab.research.google.com/drive/1NMaLS8PG0eYhbd8IxQAajXgXNIZ_AvHo?usp=sharing) shows, Nvidia DALI is about 20% faster than torchvision, even on an old K80 GPU.
+
+## Model Inference Optimizations
+
+Start model inference optimization only after other factors, the “low-hanging fruit”, have been extensively evaluated and addressed.
+
+- Use fp16 for GPU inference.  The speed will most likely more than double on newer GPUs with tensor cores, with negligible accuracy degradation.  Technically fp16 is a type of quantization but since it seldom suffers from loss of accuracy for inference it should always be explored. As shown in this [article](https://docs.nvidia.com/deeplearning/performance/mixed-precision-training/index.html#abstract), use of fp16 offers speed up in large neural network applications.
+
+- Use model quantization (i.e., int8) for CPU inference.  Explore different quantization options: dynamic quantization, static quantization, and quantization aware training, as well as tools such as Intel Neural Compressor that provide more sophisticated quantization methods.
```
Comment (msaroufim) [inline docs/performance_checklist.md:26]: add note to ensure you check this is actually fast on your hardware

File: docs/performance_checklist.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+- Try optimized inference engines such as onnxruntime, tensorRT, lightseq, ctranslate-2, etc.  These engines often provide additional optimizations such as operator fusion, in addition to model quantization.
+
+- Try model distillation.  This is more involved and often requires training data, but the potential gain can be large.  For example, MiniLM achieves 99% the accuracy of the original BERT base model while being 2X faster.
+
+- Try task parallelism.  Python’s GIL could affect effective multithreading, even for external native code.  For a system with 32 vCPUs, two inference sessions each with 16 threads often have higher throughput than a single inference session with 32 threads.  When testing multiple sessions, it is important to set torch.num_threads properly to avoid CPU contention.
```
Comment (msaroufim) [inline docs/performance_checklist.md:36]: hmm this is no longer relevant now that we have core pinning

**Review by msaroufim — CHANGES_REQUESTED** [review]
see comments inline

**Comment by sekyondaMeta** [comment]
> 

Made the updates

**Review by chauhang — APPROVED** [review]

**Pull request merged**